src/bluetooth.hpp

```diff
@@ -0,0 +1,63 @@
+#ifndef BLUETOOTH
+#define BLUETOOTH
+
+#include "wrap-hwlib.hpp"
+
+class bluetooth {
+public:
+     /// \brief
+    /// Used to connect to other devices
+    /// \details
+    /// Connect to the specified device ID. To get the device ID use function "search".
+    virtual void connect(int deviceID) = 0;
+
+    /// \brief
+    /// Used to disconnect form a device.
+    virtual void disconnect() = 0;
+
+    /// \brief
+    /// Used to read the name of this device.
+    virtual uint8_t* getName() = 0;
+
+    /// \brief
+    /// Used to read the status of the HC-06 chip.
+    virtual int getStatus() = 0;
+
+    /// \brief
+    /// Used to pair with other devices.
+    /// \details
+    /// Pair with the specified device ID. To get the device ID use function "search".
+    virtual void pair(int deviceID) = 0;
+    
+    /// \brief
+    /// Used to retrieve the message.
+    /// \details
+    /// Provide this function with own data buffer. The buffer will hold the received message.
+    virtual void receive(uint8_t *data) = 0;
+
+    /// \brief
+    /// Used to discover any discoverable devices.
+    /// \details
+    /// This function will return a device id for all discovered devices. Use this ID to connect.
+    virtual uint8_t* search() = 0;
+
+    /// \brief
+    /// Used to send data to other devices.
+    /// \details
+    /// This function will send the data provided.
+    virtual void send(uint8_t *data) = 0;
+
+    /// \brief
```

Comment:
Incorrect use of Doxygen, please read the wiki for the correct Doxygen standard.